plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/reload/ReloadPipelinesAction.java

```diff
@@ -0,0 +1,137 @@
+    public ReloadPipelinesAction(Settings settings, Provider<PipelineStore> pipelineStoreProvider, ClusterService clusterService, TransportService transportService) {
+        super(settings);
+        this.pipelineStoreProvider = pipelineStoreProvider;
+        this.clusterService = clusterService;
+        this.transportService = transportService;
+        transportService.registerRequestHandler(ACTION_NAME, ReloadPipelinesRequest::new, ThreadPool.Names.SAME, this);
+    }
+
+    public void reloadPipelinesOnAllNodes(Consumer<Boolean> listener) {
+        List<DiscoveryNode> ingestNodes = new ArrayList<>();
+        for (DiscoveryNode node : clusterService.state().getNodes()) {
+            String nodeEnabled = node.getAttributes().get("ingest");
+            if ("true".equals(nodeEnabled)) {
+                ingestNodes.add(node);
+            }
+        }
+
+        if (ingestNodes.isEmpty()) {
+            throw new IllegalStateException("There are no ingest nodes in this cluster");
+        }
+
+        AtomicBoolean failed = new AtomicBoolean();
+        AtomicInteger expectedResponses = new AtomicInteger(ingestNodes.size());
+        for (DiscoveryNode node : ingestNodes) {
+            ReloadPipelinesRequest nodeRequest = new ReloadPipelinesRequest();
+            transportService.sendRequest(node, ACTION_NAME, nodeRequest, new TransportResponseHandler<ReloadPipelinesResponse>() {
+                @Override
+                public ReloadPipelinesResponse newInstance() {
+                    return new ReloadPipelinesResponse();
+                }
+
+                @Override
+                public void handleResponse(ReloadPipelinesResponse response) {
+                    decrementAndReturn();
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    logger.warn("failed to update pipelines on remote node [{}]", exp, node);
+                    failed.set(true);
+                    decrementAndReturn();
+                }
+
+                void decrementAndReturn() {
+                    if (expectedResponses.decrementAndGet() == 0) {
+                        listener.accept(!failed.get());
+                    }
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.MANAGEMENT;
+                }
+            });
+        }
+    }
+
+    @Override
+    public void messageReceived(ReloadPipelinesRequest request, TransportChannel channel) throws Exception {
+        try {
+            pipelineStore().updatePipelines();
+            channel.sendResponse(new ReloadPipelinesResponse());
+        } catch (Throwable e) {
+            logger.warn("failed to update pipelines", e);
+            channel.sendResponse(e);
+        }
+    }
+
+    private PipelineStore pipelineStore() {
+        if (pipelineStore == null) {
+            pipelineStore = pipelineStoreProvider.get();
```

Comment:
this method is entirely unused and you don't need that `pipelineStoreProvider` at all